apps/docs/src/app/core/component-docs/shellbar/shellbar-docs.component.html

```diff
@@ -45,20 +48,101 @@
+    To extend existing use <code>fd-user-actions-menu-header</code> component.
+</description>
+<component-example>
+    <fd-shellbar-user-actions-menu-header-example></fd-shellbar-user-actions-menu-header-example>
+</component-example>
+<code-example [exampleFiles]="shellbarUserActionsMenuHeaderExample"></code-example>
+
+<separator></separator>
+
+<fd-docs-section-title id="user-actions-menu-without-header" componentName="shellbar">
+    User actions menu without header
+</fd-docs-section-title>
+<description>
+    To hide default header content use <code>fd-user-actions-menu-header</code> component with
+    <code>[showMainHeader]="false"</code>.<br />
+    To redefine existing header content use <code>fd-user-actions-menu-header</code> component with
+    <code>[showMainHeader]="false"</code> and paste new content to the <code>fd-user-actions-menu-header</code>.
+</description>
+<component-example>
+    <fd-shellbar-user-actions-menu-without-header-example></fd-shellbar-user-actions-menu-without-header-example>
+</component-example>
+<code-example [exampleFiles]="shellbarUserActionsMenuWithoutHeaderExample"></code-example>
+
+<separator></separator>
+
+<fd-docs-section-title id="user-actions-menu-header-with-addons" componentName="shellbar">
+    User actions menu header with addons
+</fd-docs-section-title>
+<description>
+    To define addons use <code>[fd-user-actions-menu-header-addon-left]</code> or
+    <code>[fd-user-actions-menu-header-addon-right]</code> directives.
+</description>
+<component-example>
+    <fd-shellbar-user-actions-menu-header-with-addons-example></fd-shellbar-user-actions-menu-header-with-addons-example>
+</component-example>
+<code-example [exampleFiles]="shellbarUserActionsMenuHeaderWithAddonsExample"></code-example>
+
+<separator></separator>
+
+<fd-docs-section-title id="user-actions-menu-header-footer" componentName="shellbar">
+    User actions menu with footer
+</fd-docs-section-title>
+<description> To define footer use <code>fd-user-actions-menu-footer</code> component. </description>
+<component-example>
+    <fd-shellbar-user-actions-menu-footer-example></fd-shellbar-user-actions-menu-footer-example>
+</component-example>
+<code-example [exampleFiles]="shellbarUserActionsMenuFooterExample"></code-example>
+
+<separator></separator>
+
+<fd-docs-section-title id="user-actions-menu-submenu" componentName="shellbar">
+    User actions menu item with submenu
+</fd-docs-section-title>
+<description>
+    To define item submenu use <code>[fd-user-actions-menu-item]</code> directive with <code>submenu</code> input.
+</description>
+<component-example>
+    <fd-shellbar-user-actions-menu-submenu-example></fd-shellbar-user-actions-menu-submenu-example>
+</component-example>
+<code-example [exampleFiles]="shellbarUserActionsMenuSubmenuExample"></code-example>
+
+<separator></separator>
+
+<fd-docs-section-title id="user-actions-menu-compact" componentName="shellbar">
+    User actions menu compact mode
```

Comment:
When clicked on ‘example click,’ ‘SubMenu level 2’ is defined and further on ‘SubMenu level 3’ appears with the sign out option being visible at the footer.